opm/autodiff/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -170,6 +174,18 @@ namespace Opm
                                 perfPress()[ perf ] = prevState.perfPress()[ oldPerf_idx ];
                             }
                         }
+                        // perfSolventRates
+
+                        if (pu.has_solvent) {
+                            if( num_perf_old_well == num_perf_this_well )
+                            {
+                                for (int perf = wells->well_connpos[ newIndex ];
+                                     perf < wells->well_connpos[ newIndex + 1]; ++perf, ++oldPerf_idx )
```

Comment:
I think there is a bug here: oldPerf_idx has already been used as a loop variable by the perfPressures loop above. The existing code was dangerous in that respect. Suggested fix: make oldPerf_idx a const variable on line 148 (new numbering) and use separate loop-scoped variables in the perfPressures and perfSolventRates loops.

Reply:
Good catch. Thanks.